operator/src/lib.rs

```diff
@@ -0,0 +1,86 @@
+mod operator;
+
+use std::collections::HashMap;
+use std::ffi::{OsStr, OsString};
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+
+use anyhow::bail;
+use clap::Args;
+use reth_primitives::Address;
+
+pub use operator::Operator;
+
+#[derive(Args)]
+pub struct Config {
+    #[arg(long)]
+    /// Path to directory containing operator keypairs
+    pub operator_keypair_path: PathBuf,
+
+    #[arg(long)]
+    /// Prefix filter for operator keys
+    pub operator_keypair_prefix: Option<OsString>,
+}
+
+#[derive(Debug, Clone)]
+pub struct Operators {
+    map: Arc<HashMap<Address, Arc<Operator>>>,
+}
+
+impl Operators {
+    pub fn from_config(config: Config) -> anyhow::Result<Self> {
+        Self::load_from_path(
+            config.operator_keypair_path,
+            config.operator_keypair_prefix.as_deref(),
+        )
+    }
+
+    pub fn load_from_path(path: impl AsRef<Path>, prefix: Option<&OsStr>) -> anyhow::Result<Self> {
+        let path = path.as_ref();
+        if !path.is_dir() {
+            bail!("provided path ({path:?}) is not a directory")
+        }
+
+        macro_rules! ok {
+            ($result:expr) => {
+                match $result {
+                    Ok(entry) => entry,
+                    Err(error) => {
+                        tracing::warn!(?error, dir = ?path, "error reading directory entry");
+                        continue;
+                    }
+                }
+            }
+        }
+
+        let mut map = HashMap::new();
+        // TODO: tokio read_dir??
```

Comment:
We can remove this TODO. We run this code in the initialisation phase, so it makes little sense to make it async.

Reply:
I prefer for this TODO to stay since I plan on introducing reload logic that will be used in async code.

Reply:
Fair enough 👍